pygfx/objects/_volume.py

```diff
@@ -0,0 +1,40 @@
+from ._base import WorldObject
+from ..geometries import BoxGeometry
+from ..resources import Buffer
+
+
+class Volume(WorldObject):
+    """A volume represents a 3D image in space. It has an implicit
+    geometry based on the shape of the texture (the map of the
+    material), and the voxel (0,0,0) will be at the origin of the local
+    coordinate frame. Positioning and dealing with anisotropy should
+    be dealt with using the scale and position properties.
+    """
+
+    def __init__(self, material):
+        super().__init__()
+        self.material = material
+
+    @property
+    def material(self):
+        """The material of the volume."""
+        return self._material
+
+    @material.setter
+    def material(self, material):
+        self._material = material
+        self._set_geometry(material.map.size)
+
+    def _set_geometry(self, size):
```

Comment:
I think it would be fine if this is a public method.

In a similar fashion, a helper method to set `self.matrix` based on position, rotation and spacing would be very nice of course.

Reply:
Good point.

> ... a helper method to set self.matrix

This is a world object, so it already has properties for position, rotation and scaling. That should be enough, right?

Reply:
done